src/pages/page2.tsx

```diff
@@ -1,11 +1,79 @@
+import { useState } from 'react';
 
-function Page2() {
+const Page2 = () => {
+  // State for plan selection and error message
+  const [selectedPlan, setSelectedPlan] = useState('');
+  const [error, setError] = useState('');
+
+  // Simple navigation to go back to Page 1
+  const handleGoBack = () => {
+    window.location.href = '/page1'; // Redirect to Page 1
+  };
+
+  const handlePlanSelection = (plan) => {
+    setSelectedPlan(plan);
+    setError(''); // Clear error when a plan is selected
+  };
+
+  const handleNextStep = () => {
+    if (!selectedPlan) {
+      setError('Please select a plan!'); // Show error if no plan is selected
+    } else {
+      // Navigate to the next page if a plan is selected
+      window.location.href = '/nextpage'; // Change to your actual next page URL
```

Comment:
you need to use useNavigation from react router dom here to navigate instead of windows.location